apisx/v1alpha2/shared_types.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	v1 "sigs.k8s.io/gateway-api/apis/v1"
+	v1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+type (
+	// +k8s:deepcopy-gen=false
+	SessionPersistence = v1.SessionPersistence
+	// +k8s:deepcopy-gen=false
+	Duration = v1.Duration
+	// +k8s:deepcopy-gen=false
+	PolicyStatus = v1alpha2.PolicyStatus
+	// +k8s:deepcopy-gen=false
+	LocalPolicyTargetReference = v1alpha2.LocalPolicyTargetReference
+)
+
+// RequestRate expresses a rate of requests over a given period of time.
+//
+// +kubebuilder:validation:XValidation:message="interval can not be greater than one hour",rule="!(duration(self.interval) == duration('0s') || duration(self.interval) > duration('1h'))"
```

Comment:
Why not add this on the interval field directly? Can you add some basic CEL tests for this as well? (https://github.com/kubernetes-sigs/gateway-api/tree/main/pkg/test/cel)

Reply:
I agree with @robscott here – I think this should go directly on the `Interval` field so that it's clear to anyone looking for the limits (or to doc gen, etc).